contents/blog/how-we-built-web-analytics.md

```diff
@@ -0,0 +1,118 @@
+---
+title: The hard parts of building a Google Analytics alternative
+date: 2024-10-24
+author:
+ - lior-neu-ner
+ - robbie-coomber
+rootpage: /blog
+featuredImage: >- 
+  https://res.cloudinary.com/dmukukwp6/image/upload/mobile_replay_7ff47733d8.png
+featuredImageType: full
+tags:
+ - Guides
+---
+
+Last month we finally launched [web analytics](/web-analytics), our Google Analytics alternative 🎉 You would think for a company that already does [product analytics](/product-analytics), adding web analytics would be easy. 
+
+Alas, this is not the case. It took <TeamMember name="Robbie Coomber" /> a year to do it, and it's not because he's lazy. 
+
+Fundametally, they have different use cases and thus require different architectures. In this post, we dive into the differences between the two, the challenges we faced, and how we solved them.
+
+![Photo of Robbie being lazy](placeholder)
+<Caption>Robbie. He's definitely not lazy.</Caption>
+
+## The difference between web and product analytics
```

Comment:
What I think this piece is missing is a strong **why**. 

- Why did we build web analytics when we already had product analytics?
- Why did we need to make it cheaper and faster?

A potential outline to fix this:

1. Why web analytics needs to be cheaper (the business case)
    - Google Analytics is free (theoretically)
    - Other low cost competitors
    - We expect users to be larger volume, with lower user LTV. More B2C, rather than B2B.
    - This also means more data to query, which users expect to be super fast. Google Analytics is fast.
    - If we just built web analytics with our existing product analytics setup, it wouldn't be fast or cheap enough for our high volume users
2.  How we made it cheaper (Making web analytics cheaper)
    - Person processing was the big cost. Maybe some inspo here: https://github.com/PostHog/posthog.com/pull/8118
    - Solution: Added anonymous events + explain
3. How we made it faster (Making web analytics faster)
    - Luckily, the queries are the same or very similar every time, we can make a lot of optimizations
    - Product analytics data is across many granules
    - Solution: Optimizing queries, session tables, granules
4. End